generator/models.ts

```diff
@@ -0,0 +1,26 @@
+enum ScopeType {
+    None = 0,
+    Unknown = 1 << 0,
+    Tenant = 1 << 1,
+    Subcription = 1 << 2,
+    ResourceGroup = 1 << 3,
+    ManagementGroup = 1 << 4,
+    Extension = 1 << 5,
```

Comment:
>Extension [](start = 4, length = 9)

and "Extension" needs to be have its own type?

Reply:
It doesn't really fit in well with any other category, and `Extension` is how it's referred to by https://github.com/Azure/autorest.azureresourceschema